cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -981,6 +984,14 @@ public void createOrDeleteReplicas(Map<Short, Set<String>> topicsByReplicationFa
         int[] cursors = new int[racks.size()];
         int rackCursor = 0;
         for (PartitionInfo partitionInfo : cluster.partitionsForTopic(topic)) {
+
+          Partition partition = partition(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
```

Comment:
Not introduced in this PR: I noticed that the same `TopicPartition` is used below within this scope. Can we extract the following here:
```
TopicPartition tp = new TopicPartition(topic, partitionInfo.partition());
```

And use this tp anywhere else needed in this scope? Reusing the same immutable object is more efficient than recreating it multiple times below.